app/models/event.rb

```diff
@@ -51,51 +51,62 @@ def to_s
   end
 
   def unavailabilities
-    responses.unavailable + partial_responses.unavailable
+    Availability.overlapping(start_time..end_time).active.unavailable
   end
 
   def unavailable_people
-    unavailabilities.map{|a| a.person}
+    Person.active.joins(:availabilities).merge(unavailabilities)
   end
 
   def partial_responses
-    Availability.partially_available(self.start_time..self.end_time).active
+    Availability.partially_overlapping(start_time..end_time).active
   end
 
   def partial_availabilities
     partial_responses.available
   end
 
   def partially_available_people
-    partial_availabilities.includes(:person).map{|a| a.person}.uniq
+    Person.active.joins(:availabilities).
+      merge(Availability.partially_overlapping(start_time..end_time).available)
   end
 
   def partial_responding_people
-    self.partial_responses.map { |a| a.person }
+    Person.active.joins(:availabilities).
+      merge(Availability.partially_overlapping(start_time..end_time))
   end
 
   def availabilities
     responses.available
   end
 
   def available_people
-    responses.includes(:person).available.map{|a|a.person}.uniq
+    Person.active.joins(:availabilities).
+      merge(Availability.containing(start_time..end_time).available)
   end
 
   def responses
-    Availability.for_time_span(self.start_time..self.end_time).active
+    Availability.containing(start_time..end_time).active
   end
 
   def responding_people
-    self.responses.map { |a| a.person }
+    Person.active.joins(:availabilities).
+      merge(Availability.containing(start_time..end_time))
   end
 
   def eligible_people
-    Person.active.where(department: self.departments).where("title_order <= ?", Person::TITLE_ORDER[min_title])
```

Comment:
This change is incorrect, it should be `<=`.
1 is the most important title. 100 would be a less important title.
 Ideally we should have a test on this, as we've had this regression before.

Reply:
Oh, so a person ranked higher than another should have a `TITLE_ORDER` that is *less than* the other person's.

Thanks for the heads up!